tests/unit/amo/components/TestAddonsByAuthorsCard.js

```diff
@@ -232,6 +232,28 @@ describe(__filename, () => {
     }));
   });
 
+  it('dispatches fetchAddonsByAuthors on mount', () => {
```

Comment:
So I think we already have such a test case. What we would like to have now is a test case that fails when you revert your change in `src/amo/components/AddonsByAuthorsCard/index.js`, i.e. trying to render the component with a list of add-ons.

Reply:
thanks for feedback @willdurand! I just need a little bit of clarification : ) Initially it was only fetching on mount if there were NO add-ons. So I am thinking maybe the following test should be removed:
```javascript
  it('should dispatch a fetch action if no add-ons found', () => {
    const { store } = dispatchClientMetadata();
    const dispatchSpy = sinon.spy(store, 'dispatch');
    const errorHandler = createStubErrorHandler();

    render({
      addonType: ADDON_TYPE_EXTENSION,
      authorUsernames: ['test2'],
      errorHandler,
      store,
    });

    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
      addonType: ADDON_TYPE_EXTENSION,
      authorUsernames: ['test2'],
      errorHandlerId: errorHandler.id,
    }));
  });
```
altho it wasn't failing it's just not need now b/c regardless of amount of add-ons it's going to do this fetch on mount - which it does on initial load and when you move around on client side (without back and forward buttons)...so it does this not only when there are no add-ons. 

When i revert my changes (for testing purposes), i don't see anything failing via tests. But i guess i should add a test to make sure one of the items doesn't have a matching slug to the current slug and make sure that all is correct. Is that what you mean?

Reply:
Right, this test case should not be kept or maybe it should be updated to "should always fetch addons by authors". That would also cover your change, because if you revert your change in the component, this test will fail.

Reply:
hmm it;s not failing as expected here when i revert the component so i need to look into this a bit more :/